app/services/time_round.rb

```diff
@@ -0,0 +1,23 @@
+class TimeRound
+  GRANULARITY = 30.minutes.freeze
+
+  class << self
+    def floor_time(time)
+      if time >= time.beginning_of_hour + GRANULARITY
```

Comment:
you can extract this condition to separate method with descriptive name and than use guard clause

```
def some_method?(time)
 time >= time.beginning_of_hour + GRANULARITY
end

def floor_time(time)
return time.beginning_of_hour + GRANULARITY if  some_method?(time)
time.beginning_of_hour
end
```